File: backend/src/main/java/com/yourorg/appname/service/DepartmentService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.response.DepartmentResponse;
import java.util.List;

public interface DepartmentService {
    List<DepartmentResponse> getAllDepartments();
    DepartmentResponse getDepartmentById(Long id);
    DepartmentResponse getDepartmentByCode(String code);
}

File: frontend/src/services/departmentService.js
import apiClient from './apiClient';
import { ENDPOINTS } from '../constants/apiEndpoints';

export const departmentService = {
  async getAllDepartments() {
    const response = await apiClient.get(ENDPOINTS.DEPARTMENTS.LIST);
    return response.data.data;
  },

  async getDepartmentById(id) {
    const response = await apiClient.get(ENDPOINTS.DEPARTMENTS.DETAIL(id));
    return response.data.data;
  },

  async getDepartmentByCode(code) {
    const response = await apiClient.get(ENDPOINTS.DEPARTMENTS.BY_CODE(code));
    return response.data.data;
  },
};

export default departmentService;
